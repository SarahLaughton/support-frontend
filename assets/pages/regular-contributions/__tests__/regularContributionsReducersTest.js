// @flow

// ----- Imports ----- //

import type { CheckoutFailureReason } from 'helpers/checkoutErrors';
import createReducer from '../regularContributionsReducer';


// ----- Tests ----- //

describe('Regular contributions Reducer', () => {

  const reducer = createReducer(20, 'DirectDebit', 'MONTHLY', 'GBPCountries');

  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle CHECKOUT_ERROR', () => {

    const insufficientFunds: CheckoutFailureReason = 'insufficient_funds';
    const action = {
      type: 'CHECKOUT_ERROR',
      checkoutFailureReason: insufficientFunds,
    };

    const newState = reducer(undefined, action);

    expect(newState.regularContrib.checkoutFailureReason).toEqual(insufficientFunds);
    expect(newState.regularContrib.paymentStatus).toMatchSnapshot();
  });

  it('should handle SET_PAYPAL_BUTTON', () => {

    const value = 'ExpressCheckout';
    const action = {
      type: 'SET_PAYPAL_BUTTON',
      value,
    };

    const newState = reducer(undefined, action);

    expect(newState.regularContrib.checkoutFailureReason).toMatchSnapshot();
  });

  it('should handle CREATING_CONTRIBUTOR', () => {

    const action = {
      type: 'CREATING_CONTRIBUTOR',
    };

    const newState = reducer(undefined, action);
    expect(newState.regularContrib.paymentStatus).toEqual('Pending');
  });

});
