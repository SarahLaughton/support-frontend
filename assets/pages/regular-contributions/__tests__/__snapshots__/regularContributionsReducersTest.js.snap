// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regular contributions Reducer should handle CHECKOUT_ERROR 1`] = `"Failed"`;

exports[`Regular contributions Reducer should handle SET_PAYPAL_BUTTON 1`] = `null`;

exports[`Regular contributions Reducer should return the initial state 1`] = `
Object {
  "csrf": Object {
    "token": null,
  },
  "directDebit": Object {
    "accountHolderConfirmation": false,
    "accountHolderName": "",
    "accountNumber": "",
    "formError": "",
    "isDDGuaranteeOpen": false,
    "isPopUpOpen": false,
    "phase": "entry",
    "sortCodeArray": Array [
      "",
      "",
      "",
    ],
  },
  "marketingConsent": Object {
    "confirmOptIn": null,
    "error": false,
  },
  "regularContrib": Object {
    "amount": 20,
    "contributionType": "MONTHLY",
    "error": null,
    "payPalHasLoaded": false,
    "paymentMethod": "DirectDebit",
    "paymentStatus": "NotStarted",
    "pollCount": 0,
    "statusUri": null,
  },
  "user": Object {
    "displayName": "",
    "email": Object {
      "shouldValidate": false,
      "value": "",
    },
    "firstName": Object {
      "shouldValidate": false,
      "value": "",
    },
    "fullName": Object {
      "shouldValidate": false,
      "value": "",
    },
    "gnmMarketing": false,
    "id": "",
    "isPostDeploymentTestUser": false,
    "isSignedIn": false,
    "isTestUser": null,
    "lastName": Object {
      "shouldValidate": false,
      "value": "",
    },
  },
}
`;
